agent-inject/agent/agent.go

```diff
@@ -182,6 +188,8 @@ func New(pod *corev1.Pod, patches []*jsonpatch.JsonPatchOperation) (*Agent, erro
 		ServiceAccountName: saName,
 		ServiceAccountPath: saPath,
 		Status:             pod.Annotations[AnnotationAgentStatus],
+		AutoAuthMethod:     pod.Annotations[AnnotationAgentAutoAuthMethod],
```

Comment:
It's a great idea to add support for other auto auth methods. IMO it's better to have AutoAuthConfig which takes json and you can set the params to not have large number of annotations. Thoughts?